hbase-server/src/test/java/org/apache/hadoop/hbase/mob/TestMobCompactionWithDefaults.java

```diff
@@ -152,49 +167,102 @@ private void loadData(int num) {
   public void tearDown() throws Exception {
     admin.disableTable(tableDescriptor.getTableName());
     admin.deleteTable(tableDescriptor.getTableName());
-    HTU.shutdownMiniCluster();
   }
 
-
+  @Test
   public void baseTestMobFileCompaction() throws InterruptedException, IOException {
+    LOG.info("MOB compaction " + description() + " started");
+    loadAndFlushThreeTimes(rows, table, famStr);
```

Comment:
This is potentially a flakey test under load. Probably do an assert of the number of MOB files/flushes before running the MOB compaction. We could retry in a loop until we reach the required number of files by waiting until the flushes are done?

Reply:
The implementation of `loadAndFlushThreeTimes` includes an assertion on the number of generated files already.

Reply:
and it looks like the implementation also waits for the flushes to be done before doing that check. (The call to `Admin.flush` is synchronous.)